types/processor.d.ts

```diff
@@ -0,0 +1,30 @@
+import { Dimensions, MaxDimensions } from './types';
+
+export type IiifImage = {
+  canonicalLink: string;
+  profileLink: string;
+  contentType: string;
+  body: Buffer;
+};
+
+export type InfoJson = {
+  contentType: 'application/json';
+  body: string;
+};
+
+export declare class Processor {
+  constructor(
+    url: string,
+    streamResolver: (id: string) => NodeJS.ReadableStream,
```

Comment:
I don't think this is quite right based on the docs: https://github.com/samvera/node-iiif?tab=readme-ov-file#stream-resolver.

The first parameter will be something like `opts: { id: string, baseUrl: string }`.

The callback will need a bit more thought. We'll have to accept the parameters as described here: https://github.com/samvera/node-iiif/blob/main/README.md#stream-resolver

Reply:
Ah, right, thanks for noticing that.